### PR TITLE
Add README badges and report code coverage

### PR DESCRIPTION
**Pull request by sameo**
* Add codecov and Travis badges
* Report code coverage back to codecov

### CODE REVIEW

**Comment by jodh-intel** [comment]
lgtm

File: .travis.yml [hunk 1]
```diff
@@ -22,3 +22,6 @@ install:
 
 script:
   - cd ${TRAVIS_BUILD_DIR} && make test
+
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
```
Comment (bergwolf) [inline .travis.yml]: Do you need to ask travis to create the coverage profiles or does the bash script automatically generate them?

Comment (jodh-intel) [reply .travis.yml]: This seems to be the canonical way to invoke codecov (and I cannot find a profile option).

Comment (bergwolf) [reply .travis.yml]: The current cov report is unknown after merging this PR. I guess we do need go test to create the coverage profiles. cc @sameo

Comment (jodh-intel) [reply .travis.yml]: Looks like it might be a permissions issue?

**Pull request merged**